File: Lecture004 Solving Pattern Questions (Part-2)/pattern10.py

class solution():
    def pattern(n):
        row=1
        while(row<=n):
            col=1
            
            while(col<=row):
                str=ord('D')-col+1
                print(chr(str),end=" ")
                str+=1
                col+=1
            print()
            row+=1

if __name__ == '__main__':
    ob=solution
    num = int(input("Enter Number = "))
    ob.pattern(num)

File: Lecture004 Solving Pattern Questions (Part-2)/pattern10.java
import java.util.Scanner;

class pattern10 {
    public static void main(String[] args) {
        int n;

        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a number: ");
        n = sc.nextInt();
        sc.close();

        for (int row = 1; row <= n; row++) {
            // print spaces(1st triangle)
            for (int col = n - row; col >= 1; col--) {
                System.out.print(" ");
            }

            // print 2nd triangle
            for (int col = 1; col <= row; col++) {
                System.out.print(col);
            }

            // print 3rd triangle
            for (int col = row - 1; col >= 1; col--) {
                System.out.print(col);
            }

            System.out.println();
        }
    }
}
